sympy/core/expr.py

```diff
@@ -299,11 +299,13 @@ def __ge__(self, other):
         n2 = _n2(self, other)
         if n2 is not None:
             return _sympify(n2 >= 0)
-        if self.is_extended_real or other.is_extended_real:
-            dif = self - other
-            if dif.is_extended_nonnegative is not None and \
-                    dif.is_extended_nonnegative is not dif.is_extended_negative:
-                return sympify(dif.is_extended_nonnegative)
+        if self.is_extended_real and other.is_extended_real:
+            if (self.is_infinite and self.is_extended_nonnegative) \
+                    or (other.is_infinite and other.is_extended_nonpositive):
```

Comment:
What about `oo > oo`?

Reply:
It would return `False`. since the first part of or is satisfied.

Reply:
This is `__ge__` so it would be `oo >= oo`. Since the first condition applies this will return True

Reply:
Oh I missed that this is >=, not >.

Reply:
So I think the logic here is correct for the different inequalities. It could probably be simplified a bit, but we can do that in a future PR. As long as the tests cover all the cases it is good as is for now.